lib/mysql.js

```diff
@@ -374,929 +315,137 @@ MySQL.prototype._serializeObject = function(obj) {
   } else {
     val = obj;
   }
-  return this.client.escape(val);
+  return val;
 };
 
 /*!
- * Convert the data from database
- * @param model
- * @param data
- * @returns {*}
+ * Convert the data from database column to model property
+ * @param {object} Model property descriptor
+ * @param {*) val Column value
+ * @returns {*} Model property value
  */
-MySQL.prototype.fromDatabase = function (model, data) {
-  if (!data) {
-    return null;
-  }
-  var props = this._models[model].properties;
-  var json = {};
-  for (var p in props) {
-    var key = this.column(model, p);
-    var val = data[key];
-    if (val === undefined) {
-      continue;
-    }
-    if (val === null) {
-      json[p] = null;
-      continue;
-    }
-    if (props[p]) {
-      switch (props[p].type.name) {
-        case 'Number':
-          val = Number(val);
-          break;
-        case 'String':
-          val = String(val);
-          break;
-        case 'Date':
-          val = new Date(val.toString().replace(/GMT.*$/, 'GMT'));
-          break;
-        case 'Boolean':
-          val = Boolean(val);
-          break;
-        case 'GeoPoint':
-        case 'Point':
-          val = {
-            lat: val.x,
-            lng: val.y
-          };
-          break;
-        case 'List':
-        case 'Array':
-        case 'Object':
-        case 'JSON':
-          break;
-        default:
-          if (!Array.isArray(props[p].type) && !props[p].type.modelName) {
-            // Do not convert array and model types
-            val = props[p].type(val);
-          }
-          break;
-      }
+MySQL.prototype.fromColumnValue = function(prop, val) {
```

Comment:
We can push some them into the base SqlConnector as we already do for buildExpression.